### PR TITLE
fix(simple_planning_simulator): fix steering bias model

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

In the old implementation, `steering_bias` was just added in the measurement, thus the behavior was "the vehicle yaw dynamics totally follow the steering command, but the steering output seems different". However, the expected behavior is "the steering output seems to be ideal, but the vehicle yaw dynamics does not match its steering angle". This PR fixes the behavior.

## Related links

None

## Tests performed

1. Add `steer_bias` param in yaml. (e.g. set to 0.01)
2. Set `steering_offset.enable_auto_steering_offset_removal` to false in mpc.param.yaml
3. Run psim
4. See the vehicle trajectory following is slightly biased to the right/left.

You can also check the steering offset topic `/control/trajectory_follower/controller_node_exe/output/estimated_steer_offset` converges to the bias value if you set the offset_removal true.

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None by default.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc.cpp [hunk 1]
```diff
@@ -127,7 +128,7 @@ Eigen::VectorXd SimModelDelaySteerAcc::calcModel(
   Eigen::VectorXd d_state = Eigen::VectorXd::Zero(dim_x_);
   d_state(IDX::X) = vel * cos(yaw);
   d_state(IDX::Y) = vel * sin(yaw);
-  d_state(IDX::YAW) = vel * std::tan(steer) / wheelbase_;
+  d_state(IDX::YAW) = vel * std::tan(steer + steer_bias_) / wheelbase_;
```
Comment (kosuke55) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc.cpp:131]: Only yaw among the states has a biased value. each of the states should be unified as to whether it is an actual value or a sensor value.

**Comment by kosuke55** [comment]
> Only yaw among the states has a biased value. each of the states should be unified as to whether it is an actual value or a sensor value.

fixed in https://github.com/autowarefoundation/autoware.universe/pull/6240/commits/e4dadc5b80fdc3266eec412535443bac680dead8

All states are actual values, and the values obtained by `getSteer()` are biased and used in the calculation of the steer_rate in `calcModel()` and published as an output report.

**Comment by kosuke55** [comment]
https://github.com/autowarefoundation/autoware.universe/assets/39142679/268f40cc-9378-453a-9295-3f541353c758

**Comment by kosuke55** [comment]
@TakaHoribe could you please check my commits? :pray:

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp [hunk 2]
```diff
@@ -67,11 +68,12 @@ double SimModelDelaySteerAccGeared::getAx()
 }
 double SimModelDelaySteerAccGeared::getWz()
 {
-  return state_(IDX::VX) * std::tan(state_(IDX::STEER)) / wheelbase_;
+  return state_(IDX::VX) * std::tan(state_(IDX::STEER) + steer_bias_) / wheelbase_;
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp:71]: @kosuke55 You can remove this change.

**Comment by kosuke55** [comment]
@TakaHoribe 

after https://github.com/autowarefoundation/autoware.universe/pull/6240/commits/c625256a3193ccad9ad625f7f0bfdb2d84bc2455

> Add steer_bias param in yaml. (e.g. set to 0.01)
Set steering_offset.enable_auto_steering_offset_removal to false in mpc.param.yaml
Run psim
See the vehicle trajectory following is slightly biased to the right/left.

https://github.com/autowarefoundation/autoware.universe/assets/39142679/06da9803-af27-486c-8990-1c10506ce150


> You can also check the steering offset topic /control/trajectory_follower/controller_node_exe/output/estimated_steer_offset converges to the bias value if you set the offset_removal true.



https://github.com/autowarefoundation/autoware.universe/assets/39142679/374a2400-5a25-4816-9824-e1992f78e86b


https://github.com/autowarefoundation/autoware.universe/assets/39142679/cf50e651-2fa4-4510-a718-987c1338a97f

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp (hunk 2 above)
Comment (kosuke55) [reply simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc_geared.cpp]: fixed 
c625256

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_vel.cpp [hunk 3]
```diff
@@ -63,10 +64,11 @@ double SimModelDelaySteerVel::getAx()
 double SimModelDelaySteerVel::getWz()
 {
   return state_(IDX::VX) * std::tan(state_(IDX::STEER)) / wheelbase_;
+  ;
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_vel.cpp:67]:
```suggestion
```

File: simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc.cpp [hunk 4]
```diff
@@ -66,10 +67,11 @@ double SimModelDelaySteerAcc::getAx()
 double SimModelDelaySteerAcc::getWz()
 {
   return state_(IDX::VX) * std::tan(state_(IDX::STEER)) / wheelbase_;
+  ;
```
Comment (TakaHoribe) [inline simulator/simple_planning_simulator/src/simple_planning_simulator/vehicle_model/sim_model_delay_steer_acc.cpp:70]:
```suggestion
```

**Pull request merged**